apps/vc-api/src/vc-api/credentials/credentials.service.ts

```diff
@@ -83,6 +85,43 @@ export class CredentialsService implements CredentialVerifier {
     return JSON.parse(await verifyCredential(JSON.stringify(vc), JSON.stringify(verifyOptions)));
   }
 
+  /**
+   * Assembles presentation from provided credentials according to definition
+   */
+  presentationFrom(
+    presentationDefinition: IPresentationDefinition,
+    credentials: VerifiableCredentialDto[]
+  ): PresentationDto {
+    const pex = new PEX();
+    // presentation should be created from selected credentials https://github.com/Sphereon-Opensource/pex/issues/91#issuecomment-1115940908
+    const { verifiableCredential, areRequiredCredentialsPresent } = pex.selectFrom(
+      presentationDefinition,
+      credentials as IVerifiableCredential[]
```

Comment:
Did you try extend `VerifiableCredentialDto` with `IVerifiableCredential`?

Reply:
Unfortunately this is impossible. `IVerifiableCredential` is union and therefor its static members are unknown